html/apps/tweetfeed/services/dataService.js

```diff
@@ -0,0 +1,44 @@
+(function(){
+    angular.module('myApp')
+    
+    .factory('dataService', 
+    function($http){
+        
+        return {
+            getTweets: getTweets
+        }
+        
+        function getTweets(queryType, queryTerm){
+            
+            // user's tweets latest first
+            var defaultUri = "/api/search.json?q=loklak";
+            var uri = "";
+            if(queryType === "search"){
+                console.log(queryTerm);
+                uri = "/api/search.json?q=" + queryTerm;
+            } else {
+                uri = defaultUri;
+            }
+            
+            var promise = $http.get(uri)
+            .then(searchCompleted)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["searchCompleted" was used before it was defined](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l24)